client/app/bundles/course/assessment/pages/AssessmentStatistics/index.tsx

```diff
@@ -0,0 +1,169 @@
+import { statisticsActions as actions } from 'course/assessment/reducers/statistics';
+import Page from 'lib/components/core/layouts/Page';
+import LoadingIndicator from 'lib/components/core/LoadingIndicator';
+import { useAppDispatch, useAppSelector } from 'lib/hooks/store';
+import toast from 'lib/hooks/toast';
+import useTranslation from 'lib/hooks/useTranslation';
+
+import MainGradesChart from './GradeDistribution/MainGradesChart';
+import SubmissionStatusMainAssessment from './SubmissionStatus/MainSubmissionChart';
+import MainSubmissionTimeAndGradeStatistics from './SubmissionTimeAndGradeStatistics/MainSubmissionTimeAndGradeStatistics';
+import DuplicationHistoryStatistics from './DuplicationHistoryStatistics';
+import { getAssessmentStatistics } from './selectors';
+import StudentMarksPerQuestionTable from './StudentMarksPerQuestionTable';
+
+const translations = defineMessages({
+  statistics: {
+    id: 'course.assessment.statistics.statistics',
+    defaultMessage: 'Statistics',
+  },
+  header: {
+    id: 'course.assessment.statistics.header',
+    defaultMessage: 'Statistics for {title}',
+  },
+  fetchFailure: {
+    id: 'course.assessment.statistics.fail',
+    defaultMessage: 'Failed to fetch statistics.',
+  },
+  fetchAncestorsFailure: {
+    id: 'course.assessment.statistics.ancestorFail',
+    defaultMessage: 'Failed to fetch past iterations of this assessment.',
+  },
+  fetchAncestorStatisticsFailure: {
+    id: 'course.assessment.statistics.ancestorStatisticsFail',
+    defaultMessage: "Failed to fetch ancestor's statistics.",
+  },
+  duplicationHistory: {
+    id: 'course.assessment.statistics.duplicationHistory',
+    defaultMessage: 'Duplication History',
+  },
+  marksPerQuestion: {
+    id: 'course.assessment.statistics.marksPerQuestion',
+    defaultMessage: 'Marks Per Question',
+  },
+  gradeDistribution: {
+    id: 'course.assessment.statistics.gradeDistribution',
+    defaultMessage: 'Grade Distribution',
+  },
+  submissionTimeAndGrade: {
+    id: 'course.assessment.statistics.submissionTimeAndGrade',
+    defaultMessage: 'Submission Time and Grade',
+  },
+  includePhantom: {
+    id: 'course.assessment.statistics.includePhantom',
+    defaultMessage: 'Include Phantom Student',
+  },
+});
+
+const AssessmentStatisticsPage: FC = () => {
+  const { t } = useTranslation();
+  const [tabValue, setTabValue] = useState('marksPerQuestion');
+  const [includePhantom, setIncludePhantom] = useState(false);
+
+  const { assessmentId } = useParams();
+  const parsedAssessmentId = parseInt(assessmentId!, 10);
+  const dispatch = useAppDispatch();
+
+  const statistics = useAppSelector(getAssessmentStatistics);
+
+  useEffect(() => {
+    dispatch(actions.reset());
+    dispatch(fetchAssessmentStatistics(parsedAssessmentId)).catch(() =>
+      toast.error(t(translations.fetchFailure)),
+    );
+  }, [dispatch]);
+
+  const tabComponentMapping = {
```

Comment:
I don't think it can be done. some of the mapped components here (3 of them) relies on the variable `includePhantom`, which defines a state. It needs to be inside the component

Reply:
yes, then pass includePhantom as a props to the const

Reply:
done